docs/utilities/parser.md

```diff
@@ -2,168 +2,85 @@
 
-handler(event=payload, context=LambdaContext())
-handler(event=json.dumps(payload), context=LambdaContext()) # also works if event is a JSON string
+```python title="parser_function.py" hl_lines="2 12"
+--8<-- "examples/parser/src/parser_function.py"
 ```
 
-Alternatively, you can automatically extract the model from the `event` without the need to include the model parameter in the `event_parser` function.
 
-```python hl_lines="23 24"
-    --8<-- "examples/parser/src/using_the_model_from_event.py"
-```
+**Should I use parse() or @event_parser? 🤔**
 
-#### parse function
-
-Use this standalone function when you want more control over the data validation process, for example returning a 400 error for malformed payloads.
-
-```python hl_lines="21 31" title="Using standalone parse function for more flexibility"
-from aws_lambda_powertools.utilities.parser import parse, BaseModel, ValidationError
-from typing import List, Optional
-
-class OrderItem(BaseModel):
-	id: int
-	quantity: int
-	description: str
-
-class Order(BaseModel):
-	id: int
-	description: str
-	items: List[OrderItem] # nesting models are supported
-	optional_field: Optional[str] = None # this field may or may not be available when parsing
-
-
-payload = {
-	"id": 10876546789,
-	"description": "My order",
-	"items": [
-		{
-			# this will cause a validation error
-			"id": [1015938732],
-			"quantity": 1,
-			"description": "item xpto"
-		}
-	]
-}
-
-def my_function():
-	try:
-		parsed_payload: Order = parse(event=payload, model=Order)
-		# payload dict is now parsed into our model
-		return parsed_payload.items
-	except ValidationError:
-		return {
-			"status_code": 400,
-			"message": "Invalid order"
-		}
-```
+The `parse()` function offers more flexibility and control:
 
-#### Primitive data model parsing
+- It allows parsing different parts of an event using multiple models.
+- You can conditionally handle events before parsing them.
+- It's useful for integrating with complex workflows where a decorator might not be sufficient.
+- It provides more control over the validation process.
 
-The parser allows you parse events into primitive data types, such as `dict` or classes that don't inherit from `BaseModel`. The following example shows you how to parse a [`Union`](https://docs.pydantic.dev/latest/api/standard_library_types/#union):
+The `@event_parser` decorator is ideal for:
 
-```python
---8<-- "examples/parser/src/multiple_model_parsing.py"
-```
+- Fail-fast scenarios where you want to immediately stop execution if the event payload is invalid.
+- Simplifying your code by automatically parsing and validating the event at the function entry point.
 
 ### Built-in models
 
-Parser comes with the following built-in models:
+Parser provides built-in models for parsing events from AWS services. You don't need to worry about creating these models yourself - we've already done that for you, making it easier to process AWS events in your Lambda functions.
```

Comment:
Please create an example in this section.

Reply:
Example and sample event added to built-in models